psi4/src/psi4/libmints/electricfield.h

```diff
@@ -68,6 +71,49 @@ class ElectricFieldInt : public OneBodyAOInt {
 
     static Vector3 nuclear_contribution(const Vector3 &origin, std::shared_ptr<Molecule> mol);
     static SharedMatrix nuclear_contribution_to_gradient(const Vector3 &origin, std::shared_ptr<Molecule> mol);
+
+    template <typename ContractionFunctor>
+    void compute_with_functor(ContractionFunctor &functor, SharedMatrix coords);
+};
+
+class ContractOverDipolesFunctor {
+   protected:
+    /// Pointer to the resulting operator matrix
+    double **pF_;
+    /// The array of dipoles
+    double **dipoles_;
+
+   public:
+    ContractOverDipolesFunctor(SharedMatrix dipoles, SharedMatrix F) : pF_(F->pointer()), dipoles_(dipoles->pointer()) {
+        if (F->rowdim() != F->coldim()) throw PSIEXCEPTION("Invalid Fock matrix in ContractOverCharges");
+        if (dipoles->coldim() != 3) throw PSIEXCEPTION("Dipole matrix must have 3 columns.");
+        int nbf = F->rowdim();
+        std::fill_n(pF_[0], nbf * nbf, 0.0);
```

Comment:
What if the user wants to add this contribution to an existing matrix? Maybe this function shouldn't zero it out first. Let it be up to the caller to zero it out, if desired.

I say remove it and let the user decide what they want to do. Maybe add documentation saying contributions are added to the `F`.

Reply:
Very good point! Resolved.